### PR TITLE
PIM-5170: Cascade detach on detachAll

### PR DESCRIPTION
**Pull request by fitn**
The description of what the PR does

| Q                 | A
| ----------------- | ---
| Added Specs       | N
| Added Behats      | N
| Travis CI is ok   | Y
| Changelog updated | Y

### CODE REVIEW

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function detachAll(array $objects)
             $firstObject = current($objects);
```
Comment (BitOne) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: This 2 lines could be removed now.

**Comment by BitOne** [comment]
GTM when comment fixed.

**Comment by fitn** [comment]
I removed the whole to only keep the foreach loop as the count became useless too ;)
I let the Core team check behat scenarios and merge this PR.

**Comment by jjanvier** [comment]
GTM

**Pull request merged**